crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -0,0 +1,520 @@
+        if self.filesystem != other.filesystem {
+            return Err(ErrorModel::bad_request(
+                "Storage Profile `filesystem` cannot be updated to prevent data loss.",
+                "InvalidFileSystem",
+                None,
+            )
+            .into());
+        }
+
+        if self.key_prefix != other.key_prefix {
+            return Err(ErrorModel::bad_request(
+                "Storage Profile `key_prefix` cannot be updated to prevent data loss.",
+                "InvalidKeyPrefix",
+                None,
+            )
+            .into());
+        }
+
+        Ok(())
+    }
+
+    #[must_use]
+    pub fn generate_catalog_config(&self, _: WarehouseIdent) -> CatalogConfig {
+        CatalogConfig {
+            defaults: HashMap::default(),
+            overrides: HashMap::default(),
+        }
+    }
+
+    #[must_use]
+    pub fn tabular_location(
+        &self,
+        namespace_id: NamespaceIdentUuid,
+        tabular_id: TabularIdentUuid,
+    ) -> String {
+        format!("{}{tabular_id}", self.namespace_location(namespace_id))
+    }
+
+    fn namespace_location(&self, namespace_id: NamespaceIdentUuid) -> String {
+        format!("{}{namespace_id}/", self.location())
+    }
+
+    fn location(&self) -> String {
+        if let Some(key_prefix) = &self.key_prefix {
+            format!(
+                "abfss://{}@{}.{}/{}/",
+                self.filesystem,
+                self.account_name,
+                self.endpoint_suffix
+                    .as_deref()
+                    .unwrap_or(DEFAULT_ENDPOINT_SUFFIX),
+                key_prefix.trim_matches('/').to_string()
+            )
+        } else {
+            format!(
+                "abfss:/{}@{}.{}/",
+                self.filesystem,
+                self.account_name,
+                self.endpoint_suffix
+                    .as_deref()
+                    .unwrap_or(DEFAULT_ENDPOINT_SUFFIX)
+                    .trim_end_matches('/'),
+            )
+        }
+    }
+
+    // Vended credentials will be async.
+    #[allow(clippy::unused_async)]
+    /// Generate the table configuration for S3.
+    ///
+    /// # Errors
+    /// Fails if vended credentials are used - currently not supported.
+    pub async fn generate_table_config(
+        &self,
+        _: WarehouseIdent,
+        _: TableIdentUuid,
+        _: NamespaceIdentUuid,
+        _: &DataAccess,
+        creds: &AzCredential,
+    ) -> Result<HashMap<String, String>> {
+        let AzCredential::ClientCredentials {
+            client_id,
+            tenant_id,
+            client_secret,
+        } = creds;
+        let http_client = azure_core::new_http_client();
+        let token = azure_identity::ClientSecretCredential::new(
+            http_client,
+            self.authority_host
+                .clone()
+                .unwrap_or(Url::parse("https://login.microsoftonline.com").unwrap()),
```

Comment:
Let's double check if this is constant also for private link and special azure clouds (i.e. government)

Reply:
gov & chinese clouds differ which is why we take it as a param but I feel this is an ok default value